internal/authn/oidc/oidc_test.go

```
@@ -0,0 +1,200 @@
+			},
+			expectedError: "invalid bearer token",
+		},
+		{
+			testDescription: "when_token_is_signed_using_different_public/private_key_pairs,_returns__'invalid_bearer_token'",
+			testSetup: func() (*RemoteOidcAuthenticator, context.Context, error) {
+				privateKey, _ := generateJWTSignatureKeys()
+				return quickConfigSetup("kid_1", "kid_1", "", "", jwt.MapClaims{}, privateKey)
+			},
+			expectedError: "invalid bearer token",
+		},
+		{
+			testDescription: "when_token's_issuer_does_not_match_the_one_provided_in_the_server_configuration,_MUST_return_'invalid_issuer'_error",
+			testSetup: func() (*RemoteOidcAuthenticator, context.Context, error) {
+				return quickConfigSetup(
+					"kid_1",
+					"kid_1",
+					"right_issuer",
+					"",
+					jwt.MapClaims{
+						"iss": "wrong_issuer",
+					},
+					nil,
+				)
+			},
+			expectedError: "invalid issuer",
+		},
+		{
+			testDescription: "when_token's_audience_does_not_match_the_one_provided_in_the_server_configuration,_MUST_return_'invalid_audience'_error",
+			testSetup: func() (*RemoteOidcAuthenticator, context.Context, error) {
+				return quickConfigSetup(
+					"kid_1",
+					"kid_1",
+					"right_issuer",
+					"right_audience",
+					jwt.MapClaims{
+						"iss": "right_issuer",
+						"aud": "wrong_audience",
+					},
+					nil,
+				)
+			},
+		},
+		{
+			testDescription: "when_the_subject_of_the_token_is_not_a_string,_MUST_return_'invalid_subject'_error",
+			testSetup: func() (*RemoteOidcAuthenticator, context.Context, error) {
+				return quickConfigSetup(
+					"kid_1",
+					"kid_1",
+					"right_issuer",
+					"right_audience",
+					jwt.MapClaims{
+						"iss": "right_issuer",
+						"aud": "right_audience",
+						"sub": 12,
+					},
+					nil,
+				)
+			},
+			expectedError: "invalid subject",
+		},
+	}
+
+	for _, testC := range testCases {
+		t.Run(testC.testDescription, func(t *testing.T) {
+			oidc, requestContext, err := testC.testSetup()
```

```suggestion
		t.Run(testC.testDescription, func(t *testing.T) {
		        if testC.expectedError == "" {
		              t.Fail("this suite is to test error cases and this test didn't have an error")
		        } 
			oidc, requestContext, err := testC.testSetup()
```

otherwise line 99 will silently pass